src/Symfony/Component/Lock/Store/SemaphoreStore.php

```diff
@@ -37,10 +37,6 @@ public static function isSupported($blocking = null)
             return false;
```

Comment:
`return extension_loaded('sysvsem');`

Reply:
Can I remove ``$blocking`` arg? Or it's BC?

Reply:
3.4 is not released yet. IMHO, you can remove it too (this method is used in the `Symfony\Component\Console\Command\LockableTrait`). ping @nicolas-grekas

Reply:
removing an argument is BC I think
(but 3.4 will be release *with* the arg isn't it? so the fact it's not released yet shouldn't matter)

Reply:
if one starts overriding/calling this method in 3.4 and we remove this argument in 4.0 without adverting in 3.4, one will keep passing/defining this argument when calling/overriding the method. Yet that would be just dead code, but if we need to add a new argument to this method in the future for whatever reason, the new argument will receive the value intended for the old arg, leading to WTF moments. Seems fragile to me

Reply:
Yes, but this component is yet in dev phasis and has never been released and according the doc:

> we promise you backward compatibility (BC) for all **minor Symfony releases**

Unless the promise is also kept in the dev stability?

Reply:
Sure, but this targets master, right? Is it planned to remove this argument from 3.4 before the release? If yes then no issue

Reply:
indeed, this argument will stay in 3.4 (because 3.x is compatible with php 5.5 and need this check) and will be released.

Reply:
Indeed, I'll open a new PR on 3.4 to remove this argument

Reply:
Ok. What about add a deprecated notice in 3.4 and remove this argument in 4.0?

Reply:
@maidmaid sounds good enough to me, as usual the deprecation should not be triggered for internal calls.

Reply:
see https://github.com/symfony/symfony/pull/24057